frontend/packages/knative-plugin/src/components/functions/CreateActionDropdown.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { history, Dropdown } from '@console/internal/components/utils';
+import { MenuAction, MenuActions, useActiveNamespace } from '@console/shared/src';
+
+export const CreateActionDropdown: React.FC = () => {
+  const { t } = useTranslation();
+  const [activeNamespace] = useActiveNamespace();
+
+  const menuActions: MenuActions = {
+    importfromGit: {
+      label: t('knative-plugin~Import from Git'),
+      onSelection: () => `/serverless-function/ns/${activeNamespace}`,
+    },
+    functionsUsingSamples: {
+      label: t('knative-plugin~Samples'),
+      onSelection: () => `/samples/ns/${activeNamespace}?sampleType=Serverless function`,
+    },
+  };
+
+  const items = menuActions
+    ? Object.keys(menuActions).reduce<Record<string, string>>((acc, key) => {
+        const menuAction: MenuAction = menuActions[key];
+        const label =
+          menuAction.label ||
+          (menuAction.model?.labelKey ? t(menuAction.model.labelKey) : menuAction.model?.label);
+        if (!label) return acc;
+
+        return {
+          ...acc,
+          [key]: label,
+        };
+      }, {})
+    : undefined;
```

Comment:
The object above have well known actions with translated `label` and `onSelection` callbacks. So the code to use `labelKey` never runs. Also the `if (!label)` statement should never be true. It looks to me that the `items` object will be the same as `menuActions`, or?

Reply:
menuActions is object with Key and Menu actions and the key is used while selecting the option from the dropdown.  While opening the dropdown component also the shared component expecting key and label. I have removed code wrt  labelKey.